tripy/tests/performance/test_perf.py

```diff
@@ -137,3 +152,15 @@ def func({arg_str}):
     # Ensure all deltas are within a few microseconds of each other
     average_delta = sum(deltas) / float(len(deltas))
     assert all(abs(delta - average_delta) < 10 for delta in deltas)
+
+
+def test_tripy_param_update():
+    m = tp.Module()
+    m.param = tp.Parameter([1, 2, 3, 4])
+
+    def measure_thunk():
+        m.param = tp.Parameter([5, 6, 7, 8])
+
+    # Since all the shapes are statically known, no compilation should need to happen
+    # and this should impose practically no overhead.
+    assert run_timed_trials(measure_thunk) < 60.0
```

Comment:
I think we want this to be a regression test - we can use the `benchmark` fixture to achieve that.

Reply:
Very handy, I didn't realize it was built into pytest.